jetbrains-rider/tst/software/aws/toolkits/jetbrains/services/lambda/execution/local/DotNetLocalLambdaRunConfigurationTest.kt

```diff
@@ -0,0 +1,96 @@
+import software.aws.toolkits.jetbrains.settings.SamSettings
+
+class DotNetLocalLambdaRunConfigurationTest : LambdaRunConfigurationTestBase() {
+
+    override fun getSolutionDirectoryName(): String = "SamHelloWorldApp"
+
+    @Test
+    fun testHandler_ValidHandler() {
+        preWarmSamVersionCache(SamSettings.getInstance().executablePath)
+        preWarmLambdaHandlerValidation(handler = defaultHandler)
+
+        runInEdtAndWait {
+            val runConfiguration = createHandlerBasedRunConfiguration(
+                handler = defaultHandler
+            )
+            runConfiguration.shouldNotBeNull()
+            runConfiguration.checkConfiguration()
+        }
+    }
+
+    @Test(
+        expectedExceptions = [ RuntimeConfigurationError::class ],
+        expectedExceptionsMessageRegExp = "Cannot find handler 'HelloWorld::HelloWorld.Function::HandlerDoesNoteExist' in project.")
+    fun testHandler_NonExistingMethodName() {
+        val nonExistingHandler = "HelloWorld::HelloWorld.Function::HandlerDoesNoteExist"
+        preWarmSamVersionCache(SamSettings.getInstance().executablePath)
+        preWarmLambdaHandlerValidation(handler = nonExistingHandler)
+
+        runInEdtAndWait {
+            val runConfiguration = createHandlerBasedRunConfiguration(
+                handler = nonExistingHandler
+            )
+            runConfiguration.shouldNotBeNull()
+            runConfiguration.checkConfiguration()
+        }
+    }
+
+    @Test(
+        expectedExceptions = [ RuntimeConfigurationError::class ],
+        expectedExceptionsMessageRegExp = "Cannot find handler 'HelloWorld::HelloWorld.UnknownFunction::FunctionHandler' in project.")
+    fun testHandler_NonExistingTypeName() {
+        val nonExistingHandler = "HelloWorld::HelloWorld.UnknownFunction::FunctionHandler"
+        preWarmSamVersionCache(SamSettings.getInstance().executablePath)
+        preWarmLambdaHandlerValidation(handler = nonExistingHandler)
+
+        runInEdtAndWait {
+            val runConfiguration = createHandlerBasedRunConfiguration(
+                handler = nonExistingHandler
+            )
+            runConfiguration.shouldNotBeNull()
+            runConfiguration.checkConfiguration()
+        }
+    }
+
+    @Test(
+        expectedExceptions = [ RuntimeConfigurationError::class ],
+        expectedExceptionsMessageRegExp = "Cannot find handler 'Fake' in project.")
+    fun testHandler_InvalidHandlerString() {
+        val invalidHandler = "Fake"
+        preWarmSamVersionCache(SamSettings.getInstance().executablePath)
+        preWarmLambdaHandlerValidation(handler = invalidHandler)
+
+        runInEdtAndWait {
+            val runConfiguration = createHandlerBasedRunConfiguration(
+                handler = invalidHandler
+            )
+            runConfiguration.shouldNotBeNull()
+            runConfiguration.checkConfiguration()
+        }
+    }
+
+    @Test(
+        expectedExceptions = [ RuntimeConfigurationError::class ],
```

Comment:
can we do the exception assertions using assertJ's assertThatThrownBy

Reply:
Rider test framework use TestNG instead of JUnit. It has a different exceptions assertion mechanism.

Reply:
AssertJ should work with any test framework I thought

Reply:
Fixed.

Reply:
Thanks!